.github/workflows/codeql.yml

```diff
@@ -19,6 +19,9 @@ on:
   schedule:
     - cron: '39 20 * * 1' # Monday (Runs every Monday at 20:39 UTC)
 
+permissions:
```

Comment:
Add comment on what the permission is allowing for better clarity.

Reply:
Sonds, good added for you and can you please check for rest as well. https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication :heart: